cdap-examples/WikipediaPipeline/src/main/java/co/cask/cdap/examples/wikipedia/WikipediaPipelineWorkflow.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.examples.wikipedia;
+
+import co.cask.cdap.api.Predicate;
+import co.cask.cdap.api.annotation.UseDataSet;
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.lib.CloseableIterator;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.api.dataset.lib.KeyValueTable;
+import co.cask.cdap.api.workflow.AbstractWorkflow;
+import co.cask.cdap.api.workflow.AbstractWorkflowAction;
+import co.cask.cdap.api.workflow.Value;
+import co.cask.cdap.api.workflow.WorkflowContext;
+import co.cask.cdap.api.workflow.WorkflowToken;
+import co.cask.common.http.HttpRequest;
+import co.cask.common.http.HttpRequests;
+import co.cask.common.http.HttpResponse;
+import com.google.common.base.Charsets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.Map;
+
+/**
+ * Workflow for Wikipedia data pipeline
+ */
+public class WikipediaPipelineWorkflow extends AbstractWorkflow {
+
+  public static final String NAME = WikipediaPipelineWorkflow.class.getSimpleName();
+
+  @Override
+  protected void configure() {
+    setName(NAME);
+    setDescription("A workflow that demonstrates a typical data pipeline to process Wikipedia data.");
+    addMapReduce(WikipediaPipelineApp.LIKES_TO_DATASET_MR_NAME);
+    condition(new EnoughDataToProceed())
+      .condition(new IsWikipediaSourceOnline())
+        .addAction(new DownloadWikiDataAction())
+      .otherwise()
+        .addMapReduce(WikipediaPipelineApp.WIKIPEDIA_TO_DATASET_MR_NAME)
+      .end()
+      .addMapReduce(WikiContentValidatorAndNormalizer.NAME)
+      .fork()
+        .addSpark(SparkWikipediaAnalyzer.NAME)
+      .also()
+        .addMapReduce(TopNMapReduce.NAME)
+      .join()
+    .otherwise()
+    .end();
+  }
+
+  static class EnoughDataToProceed implements Predicate<WorkflowContext> {
```

Comment:
They're inner classes, so need to be static. Otherwise CDAP cannot instantiate them.